pytorch_lightning/core/lightning.py

```diff
@@ -1839,11 +1844,34 @@ def to_onnx(
 
         input_sample = self._apply_batch_transfer_handler(input_sample)
 
+        if not isinstance(input_sample, (tuple, list)):
```

Comment:
what if it's a tensor or similar (numpy array etc.)? Can we instead check for `typing.Iterable` here? Maybe explicitly have to exclude mapping here.

I think the type check here isn't that simple...

Reply:
I think `Iterable` is probably fairly safe. FWIW the PyTorch [docs](https://pytorch.org/docs/stable/onnx.html#functions) provide
> args (tuple of arguments or torch.Tensor, a dictionary consisting of named arguments (optional))

Reply:
I think Iterable doesn't fit quite well since `Tensor` is iterable while the check for tuple, list is to make sure it is not a tensor, hence the explicit mapping.

Reply:
Yes, but if I have multiple inputs (let's say two images) of the same size, I could also combine them in a tensor instead of a tuple. If we want to exclude this, we may go with Sequence.